big-calendar/src/App.js

```diff
@@ -1,23 +1,115 @@
-import logo from './logo.svg';
-import './App.css';
+import { Calendar, momentLocalizer } from "react-big-calendar";
+import moment from "moment";
+import "./App.css";
+import "react-big-calendar/lib/css/react-big-calendar.css";
+import { useState } from "react";
+
+const localizer = momentLocalizer(moment);
+const currentHour = new Date().getHours();
+const nextHour = currentHour + 1;
+const currentMinutes = new Date().getMinutes();
+const formatToHourAndMinutes = (time) => time + ":00:00.000";
+const defaultStartTime =
+  currentMinutes > 0
+    ? formatToHourAndMinutes(nextHour)
+    : formatToHourAndMinutes(currentHour);
+const defaultEndTime = formatToHourAndMinutes(nextHour + 1);
+
+function padTo2Digits(num) {
+  return num.toString().padStart(2, "0");
+}
+
+function formatDate(date) {
+  return [
+    date.getFullYear(),
+    padTo2Digits(date.getMonth() + 1),
+    padTo2Digits(date.getDate()),
+  ].join("-");
+}
+
+const formatAMPM = (timeString) => {
+  const date = timeString.substring(0, 5);
+  let hours = date.substring(0, 2);
+  let minutes = date.substring(3, 5);
+  let ampm = hours >= 12 ? "pm" : "am";
+  hours = hours % 12;
+  hours = hours ? hours : 12;
+  let strTime = hours + ":" + minutes + " " + ampm;
+  return strTime;
+};
 
 function App() {
+  const [startDate, setStartDate] = useState(formatDate(new Date())); // may set this to a Date object later.
+  const [endDate, setEndDate] = useState(formatDate(new Date()));
+  const [startTime, setStartTime] = useState(defaultStartTime);
+  const [endTime, setEndTime] = useState(defaultEndTime);
+  const [title, setTitle] = useState("");
+  const [events, setEvents] = useState([]);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setEvents([
+      {
+        title: `${formatAMPM(startTime)} ${title}`,
+        start: new Date(`${startDate}T${startTime}`),
+        end: new Date(`${endDate}T${endTime}`),
```

Comment:
We need validation on the `start` and `end` values that `end` is not before `start`.

Reply:
We should be able to use [setHours](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setHours) instead, to simplify this logic and to not have to hardcode `T`.

Reply:
Following up on this - `setHours` would work if we were using DateTime objects in our state directly, but we actually use a pre-formatted date object instead. The reason we use that pre-formatted date is for easier interoperability with the calendar UI (so that we can set the dates easily as the user picks new ones, instead of having to change to a Date object and back to a date string to be able to display it in the UI).